### DISCUSSION TITLE
One colour backghround when using `defaultformatter = "syntaxhighlighting"`

### DISCUSSION THREAD

**Discussion by TehPeGaSuS** [PrivateBin/PrivateBin#1375] [Q&A] [answered]
### Describe the problem/question

So, I've been trying to understand what/where to change so I can have only one background color when using `defaultformatter = "syntaxhighlighting"`.

I honestly don't like how it looks like and had some other people complaining that it makes it harder to read the pasted code since it's kinda distracting.

I'd be super appreciated if someone could provide me a fix, so I could have a one color background, like we have when we use `defaultformatter = "plaintext"`

Thanks in advance!

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

1. Go to [my instance](https://pastebin.0bin.xyz) and paste something like the code below and hit `Create` (or even `Preview` only):
```
set {
	options {
		hide-ulines;
	}
	auto-join "#something";
}
```
2. Each line is interspersed between a kind of dark gray and a kind of less dark gray.
3. When large amounts of code are pasted (like configs, modules/plugins/etc) it is very distracting and unpleasant to the eye

### What happens

Each line is interspersed between a kind of dark gray and a kind of less dark gray.

### What should happen

A uniform background like we have when using `defaultformatter = "plaintext"`.

### Server address

https://pastebin.0bin.xyz

### Server OS

Ubuntu 22.04.4 LTS

### Webserver

Apache

### PrivateBin version

1.7.3

### Browser and version

Google Chrome Version 126.0.6478.126 (Official Build) (64-bit)

### Local operating system and version

Xubuntu (Ubuntu 22.04.4 LTS)

### Issue reproducibility

Yes, reproducible on https://privatebin.net. (although the pastebin color there is white and is a bit less unpleasant)

**Comment by elrido** [comment]
Have a look at the [syntaxhighlightingtheme configuration option](https://github.com/PrivateBin/PrivateBin/wiki/Configuration#syntaxhighlightingtheme). Apart from the default, there is also desert, doxy, sons-of-obsidian and sunburst. Though AFAIK they all have different backgrounds every other line. Maybe you find one that is a little less stark in contrast?

**Comment by TehPeGaSuS** [comment]
I did take a look but what I'd really like is to have a one color background, which isn't possible with any of the current themes.

And, as recommended by PrivateBin, since I'm using the `bootstrap-dark` template I *should* use `sons-of-obsidian` as `syntaxhighlightingtheme`.

It'd be really awesome if we could have a theme that behaves like `plaintext` even if we're using `syntaxhighlighting` on the `syntaxhighlightingtheme` setting.

**Reply by elrido** [reply]
Ok, in that case you will have to create a custom template which includes an additional CSS file with your override. You will want a selector similar to this one, to set just the odd lines and to some color that matches the even lines, for example:
```css
li.L1, li.L3, li.L5, li.L7, li.L9 {
	background-color: #000 !important;
}
```

**Reply by TehPeGaSuS** [reply]
I'll try to learn how to do all of that, since I have 0 experience with CSS and such.

I think this would be a good addition to the core, though.

Thanks!

**Comment by TehPeGaSuS** [comment]
Is there anyone willing to tell me what to change so I can achieve one colour background on **Source Code**?

I don't understand what to change and I've messed up a few times already.

Thanks in advance!

**Comment by TehPeGaSuS** [comment, marked as answer]
I managed to do it (for the bootstrap5 dark theme) by changing [this line](https://github.com/PrivateBin/PrivateBin/blob/master/css/bootstrap5/privatebin.css#L45) from `background-color: var(--bs-gray-dark) !important;` to `background-color: var(--bs-gray-900) !important;`.
